xdsl/dialects/ltl.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+from xdsl.dialects.builtin import ContainerOf
+from xdsl.ir import Attribute, SSAValue
+
+from xdsl.ir.core import Attribute, ParametrizedAttribute, TypeAttribute
+from xdsl.irdl import (
+    IRDLOperation,
+    irdl_attr_definition,
+    irdl_op_definition,
+)
+
+from typing import Annotated
+from xdsl.irdl import (
+    IRDLOperation,
+    irdl_attr_definition,
+    irdl_op_definition,
+    operand_def,
+    AnyOf,
+    ConstraintVar
+
+)
+
+from xdsl.dialects.arith import signlessIntegerLike
+from xdsl.irdl.irdl import result_def
+
+
+
+"""
+Implementation of the LTL dialect by CIRCT. Documentation: https://circt.llvm.org/docs/Dialects/LTL/
+"""
+
+
+@irdl_attr_definition
+class property(ParametrizedAttribute, TypeAttribute):
+    """
+    The `ltl.property` type represents a verifiable property built from linear
+    temporal logic sequences and quantifiers, for example, *"if you see sequence
+    A, eventually you will see sequence B"*.
+    Note that this type explicitly identifies a *property*. However, a boolean
+    value (`i1`) or a sequence (`ltl.sequence`) is also a valid property.
+    Operations that accept a property as an operand will use the `AnyProperty`
+    constraint, which also accepts `ltl.sequence` and `i1`.
+    """
+
+    name = "ltl.propertytype"
+
+
+@irdl_attr_definition
+class sequence(ParametrizedAttribute, TypeAttribute):
```

Comment:
```suggestion
class Sequence(ParametrizedAttribute, TypeAttribute):
```
Please use PascalCase for class names